web-app/HeimScripts/heatmap/run.R

```diff
@@ -176,3 +189,85 @@ add.subset.label <- function(df,label){
   }
   return(df)
 }
+
+computeDendrogram <- function(zScoreMatrix, distanceMeassure, linkageMethod) {
+    as.dendrogram(hclust(dist(zScoreMatrix, method=distanceMeassure), method=linkageMethod))
+}
+
+dendrogramToJSON <- function(d) {
+    totalMembers <- attributes(d)$members
+    add_json <- function(x, start, left) {
+        members <- attributes(x)$members
+        height <- attributes(x)$height
+        index <- (start - 1):(start + members - 2)
+        index <- paste(index, collapse=' ')
+        jsonString <<- paste(jsonString, sprintf('{"height":"%s", "index":"%s", "children":[', height, index))
+        if (is.leaf(x)){
+            jsonString <<- paste(jsonString, ']}')
+        } else {
+            add_json(x[[1]], start, TRUE)
+            jsonString <<- paste(jsonString, ",")
```

Comment:
I thought the "parent scope" was not a lexical concept and that, for instance, when you call `add_json` recursively, the parent scope of the recursive call is the scope associated with the frame below.